File: backend-dotnet/src/TaskBoard.Api/Models/TaskItem.cs
namespace TaskBoard.Api.Models;

/// <summary>
/// Entity mapped onto the existing <c>tasks</c> table. Column names are set in
/// <see cref="Data.TaskBoardContext"/>; the database owns the schema.
/// </summary>
public class TaskItem
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string Status { get; set; } = TaskStatuses.Todo;
    public string? Assignee { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

public static class TaskStatuses
{
    public const string Todo = "todo";
    public const string InProgress = "in-progress";
    public const string Done = "done";

    public static readonly IReadOnlySet<string> All =
        new HashSet<string> { Todo, InProgress, Done };

    public static bool IsValid(string status) => All.Contains(status);
}

File: backend-java/src/main/java/com/honeywell/taskboard/model/TaskItem.java
package com.honeywell.taskboard.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import org.hibernate.annotations.Generated;
import org.hibernate.generator.EventType;

/**
 * JPA entity mapped onto the existing {@code tasks} table. This is a mapping,
 * not a schema definition — database/schema.sql owns the schema and
 * {@code spring.jpa.hibernate.ddl-auto} is {@code none}.
 */
@Entity
@Table(name = "tasks")
public class TaskItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private String title;

    @Column
    private String description;

    @Column(nullable = false)
    private String status = TaskStatuses.TODO;

    @Column
    private String assignee;

    // Filled by the database (DEFAULT now()); @Generated makes Hibernate read
    // the value back after insert instead of leaving it null.
    @Generated(event = EventType.INSERT)
    @Column(name = "created_at", insertable = false, updatable = false)
    private LocalDateTime createdAt;

    // DEFAULT now() on insert, and bumped by the BEFORE UPDATE trigger.
    @Generated(event = {EventType.INSERT, EventType.UPDATE})
    @Column(name = "updated_at", insertable = false, updatable = false)
    private LocalDateTime updatedAt;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getAssignee() {
        return assignee;
    }

    public void setAssignee(String assignee) {
        this.assignee = assignee;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
}
